src/components/Garden.js

```diff
@@ -1,136 +1,162 @@
+          },
+          {
+            color: colorsByLanguage[d.languages[1]] || 'none',
+            path: petalPaths[1],
+            petalRotationStart: petalRotationStarts[1],
+            scale: flowerSizeScale(d.lifespan)
+          },
+          {
+            color: colorsByLanguage[d.languages[0]] || 'none',
+            path: petalPaths[0],
+            petalRotationStart: petalRotationStarts[0],
+            scale: flowerSizeScale(d.lifespan)
+          },
+        ];
+      })
+      .enter().append('g')
+      .attr('class', 'petal-layer')
+
+    const petal = petalLayer.selectAll('.petal')
+      .data(d => [d,d,d])
+      .enter().append('path')
+      .attr('class', 'petal')
+      .attr('fill', d => d.color)
+      .attr('d', d => d.path)
+      .attr('transform', (d,i) => `rotate(${d.petalRotationStart + i * 120 || 0})scale(${d.scale})`)
 
     const flowerCenter = flowerBed.selectAll('circle')
       .data(repositories).enter().append('circle')
-      .attr('r', 25)
+      .attr('r', 15)
       .attr('cx', (d, i) => 100 + i * 200)
       .attr('cy', d => yStemScale(d.lifespan) - 20)
       .attr('stroke-width', 1)
       .attr('stroke', 'blue')
       .attr('fill', 'blue')
-      //Append the month names to each slice
-
-
-  //   console.log('STEMS', stem)
-  //
-  //   //DRAW ROOTS
-  //   //one root for each branch
-  //   //text path w branch name
-  //   // const root =
-  //   // console.log('ROOTS', roots)
-  }
-  //
+
+      flowerBed.selectAll('.repoName')
+        .data(repositories).enter()
+        .append('text')
+        .append('textPath')
+        .attr('xlink:href', (d, i) => `#myStem${i}`)
+        .text(d => d.name)
+        .attr('font-size', '1.5rem')
+
+      const rootBox = flowerBed.selectAll('.root-box')
+        .data(repositories).enter().append('svg')
+        .attr('class', '.root-box')
+        .attr('height', '200')
+        .attr('width', '200')
+        .attr('viewBox','-150 -150 300 300')
+        .attr('x', (d, i) => i * 200)
+        .attr('y', 500)
+
+      const root = rootBox.selectAll('.root')
+        .data(d => d.branches.map(branch => {
+          return {
+            repo: d.name,
+            name: branch,
+            rotationFactor: d.branches.length === 1 ? -280 : 160 / d.branches.length
+          }
+        }))
+        .enter().append('path')
+        .attr('class', 'root')
+        .attr('fill', 'green')
+        .attr('stroke', 'green')
+        .attr('id', (d,i) => `${d.name + i}`)
+        .attr('d', () => {
+          const arrayOfLengths = [0, 110, 130, 150, 170, 180, 190, 210, 250];
+          const arrayOfCurves = [-20, -5, 5, 10, 15, 20, 25];
+          const index = (array) => [Math.floor(Math.random() * array.length)];
+          return `M0,0 C-10,40 ${arrayOfCurves[index(arrayOfCurves)]},100 0,${arrayOfLengths[index(arrayOfLengths)]}`;
+        })
+        .attr('transform', (d,i) => `rotate(${260 + (i + 1) * d.rotationFactor})`)
```

Comment:
So creative. Really, really cool. Why is the first element on the arrayOfLengths 0?

Reply:
I think that's a typo?

Reply:
I think we can consider a similar randomization of stem paths, but in that case we'll have to worry about the animation too

Reply:
another way of doing this that might be prettier: design 3+ full root paths that we store in an array and then randomly select entire path from that array